### PR TITLE
revisit instant fixup by renaming and adding variants 

### PR DESCRIPTION
**Pull request by atusy**
- Rename
    - `fixup:instant` -> `fixup:instant-fixup`
- Add
    - `fixup:instant-amend`
    - `fixup:instant-reword`

## Summary by CodeRabbit

- **Refactor**
	- Improved the fix-up functionality with a more flexible approach, including dynamic command definition and enhanced handling for instant squash actions.
	- Replaced specific functions with a more generalized approach.
	- Consolidated logic for instant squash into a separate function.
	- Updated the `doFixup` function to handle instant squash based on a new parameter.

### CODE REVIEW

File: denops/gin/action/fixup.ts [hunk 1]
```diff
@@ -17,27 +17,34 @@ export async function init(
...
-    );
     await define(
       denops,
       bufnr,
       "fixup:instant-fixup",
       (denops, bufnr, range) =>
         doFixup(denops, bufnr, range, gatherCandidates, true),
     );
+
+    const kinds: Array<"amend" | "reword"> = ["amend", "reword"];
```
Comment (lambdalisue) [inline denops/gin/action/fixup.ts:28]: Use `as const` in this case.

```
const kinds = ["amend", "reword"] as const;
```

**Review by lambdalisue — APPROVED** [review]
LGTM

**Review by lambdalisue — CHANGES_REQUESTED** [review]
Approved by miss

File: denops/gin/action/fixup.ts (hunk 1 above)
Comment (atusy) [reply denops/gin/action/fixup.ts]: Ah hah! Thanks a lot.

**Review by lambdalisue — APPROVED** [review]

**Pull request merged**